crates/miden-lib/asm/miden/note.masm

```diff
@@ -32,19 +32,14 @@ const.ERR_PROLOGUE_NOTE_INPUTS_LEN_EXCEEDED_LIMIT="number of note inputs exceede
 #! - num_inputs is greater than 1024.
 #!
 #! Invocation: exec
-export.compute_inputs_commitment
+pub proc compute_inputs_commitment
     # check that number of inputs is less than or equal to 1024
     dup.1 push.1024 u32assert2.err=ERR_PROLOGUE_NOTE_INPUTS_LEN_EXCEEDED_LIMIT
     u32lte assert.err=ERR_PROLOGUE_NOTE_INPUTS_LEN_EXCEEDED_LIMIT
     # => [inputs_ptr, num_inputs]
 
-    # push 1 as the pad_inputs flag: we should pad the stack while computing the note inputs
-    # commitment
-    push.1 movdn.2
-    # => [inputs_ptr, num_inputs, pad_inputs_flag]
-
-    exec.rpo::prepare_hasher_state
-    exec.rpo::hash_memory_with_state
+    # compute the inputs commitment
+    exec.rpo256::pad_and_hash_elements
```

Comment:
Maybe worth leaving a `TODO` here to indicate that this will be refactored as a part of https://github.com/0xMiden/miden-base/issues/2036 and/or https://github.com/0xMiden/miden-base/issues/2129.